elastic/kibana
# [Upgrade Assistant] Use skipFetchFields when creating the indexPattern in order to avoid errors if index doesn't exist

### Summary
Currently in UA when enabling the deprecation logs toggle in the Overview page and the deprecation logs data stream wasn't created yet (fresh installation or no deprecation logs have been indexed yet) the user will see a few error toasts pop up on their screen saying that the index doesn't exist. This happens because we need to have an `indexPattern` for the logs index in order to be able to construct a link to the Discovery app.

The problem originated because the method for creating the `indexPattern` tries to also fetch the index fields in order to replicate them in the indexPattern. But since the index doesn't exist yet this would throw an error that doesn't bubble up and cannot be caught from the consumer code. The solution to the problem was to not only use [allowNoIndex](https://github.com/elastic/kibana/blob/7.x/x-pack/plugins/upgrade_assistant/public/application/components/overview/fix_logs_step/external_links.tsx#L38) when creating the indexPattern, but to also use [skipFetchFields](https://github.com/elastic/kibana/blob/7.x/src/plugins/data/common/data_views/data_views/data_views.ts#L564).

**How to test**

1. Start up elasticsearch, `yarn es snapshot --license=trial`
3. Start up Kibana, `yarn start`.
4. Navigate to UA, enable deprecation logging collection
5. Verify that there are no error toasts shown on the screen and that the Discovery app link works.
